### PR TITLE
Les 36

### PR DESCRIPTION
**Pull request by evrusha**

### CODE REVIEW

**Review by fooooxmr — APPROVED** [review]

**Review by ritamma — APPROVED** [review]

File: app/models/user.rb [hunk 1]
```diff
@@ -29,4 +35,16 @@ def self.from_omniauth(auth)
       user.avatar_url = auth.info.image
     end
   end
+
+  # Validation for email: symbol . (dot) provided that it is neither the first nor the last,
+  # and also if it is not repeated more than once in a row.
+  def email_dots
+    login = email.scan(/\S+@/).join
+    errors.add(:email, 'has too many dots') if login.count('.') > 1 || login[0] == '.' || login[-2] == '.'
```
Comment (Terapool) [inline app/models/user.rb:43]: For some emails it is ok to be like 

first_name.last_name@site.com

But this is a question of the requirements

**Review by Terapool — APPROVED** [review]

**Review by Terapool — APPROVED** [review]

**Review by Dmitryshl — APPROVED** [review]

**Review by olegbrest1 — APPROVED** [review]

**Review by ritamma — APPROVED** [review]

**Review by Dmitryshl — APPROVED** [review]

File: spec/requests/sign_up_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        type: :object,
+        properties: {
+          name: { type: :string },
+          phone: { type: :string },
+          gender: { type: :string, enum: %w[male female other] },
+          email: { type: :string },
+          birthday: { type: :string },
+          password: { type: :string }
+        },
+        required: %w[id name gender email password]
```
Comment (Terapool) [inline spec/requests/sign_up_spec.rb:18]: required: %w[id ...

There is no _id_ in the properties.

File: spec/requests/sign_up_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      tags 'Sign_up'
+      produces 'application/json'
+      parameter name: :signup, in: :body, schema: {
+        type: :object,
+        properties: {
+          name: { type: :string },
+          phone: { type: :string },
+          gender: { type: :string, enum: %w[male female other] },
+          email: { type: :string },
+          birthday: { type: :string },
```
Comment (Terapool) [inline spec/requests/sign_up_spec.rb:15]: Could you leave an exampe for this request as it is not clear what format for the date is used?

File: spec/requests/sign_up_spec.rb (hunk 2 above)
Comment (evrusha) [reply spec/requests/sign_up_spec.rb:18]: corrected

File: spec/requests/sign_up_spec.rb (hunk 3 above)
Comment (evrusha) [reply spec/requests/sign_up_spec.rb]: Added date format example

**Pull request merged**